src/ng/q.js

```diff
@@ -299,14 +299,18 @@ function qFactory(nextTick, exceptionHandler, errorOnUnhandledRejections) {
    * @returns {Deferred} Returns a new instance of deferred.
    */
   var defer = function() {
-    var d = new Deferred();
-    //Necessary to support unbound execution :/
-    d.resolve = simpleBind(d, d.resolve);
```

Comment:
I think this is a small breaking change. Before, it was possible to intercept all calls to `Deferred.prototype.resolve` by changing the prototype of `Deferred` (that you can do by getting the prototype from an instance).

I think this is a small edge case, but should be clarified and the commit should make this clear.